.github/workflows/integration.yml

```diff
@@ -69,4 +69,40 @@ jobs:
         run: yarn run lerna bootstrap
 
       - name: Test
-        run: yarn run test:tool:course
+        run: yarn run test:unit:tool:course
+
+  test-e2e-course-authoring:
+      runs-on: ${{ matrix.os }}
+      strategy:
+        matrix:
+          os: [macos-latest]
```

Comment:
Is there a reason we are using macos for testing? Ubuntu should be much faster and wont need a matrix or anything like that

Reply:
Using the matrix allows us to expand what the tests run on in the future, i.e. [macos-latest, ubuntu-latest, windows-latest].

The reason we are using macos is because that is the only environment that we are building the app in. I don't think it makes much sense to test on ubuntu but build in macos.

Reply:
But the OS is not relevant in this test or for unit tests, is it? because it will run slower

Reply:
If the intention to test the executable then yeah that makes sense 👍 
but until then maybe we can use faster runner?

Reply:
Speed doesn't really matter, its an automated test running in github. If the test start taking 20mins to run locally, then we should start caring about performance.

Considering we build for macos and using electron apis that speak to it, I would say yes, the unit test will care about the environment its being tested on.